src/contracts/core/AVSDirectory.sol

```diff
@@ -344,6 +349,128 @@ contract AVSDirectory is
         }
     }
 
+    struct MagnitudeAllocation {
+        IStrategy strategy;
+        uint64 expectedTotalMagnitude;
+        OperatorSet[] operatorSets;
+        uint64[] magnitude;
+    }
+
+    function setAllocations(
+        address operator,
+        MagnitudeAllocation[] calldata allocations,
+        SignatureWithSaltAndExpiry calldata allocatorSignature
+    ) external {
+        // completable timestamp for deallocations
```

Comment:
verify msg.sender

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/691/commits/db66da58945da71f7f53940f6dbf656a7d25d61e

Reply:
still need to handle signature

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/691/commits/93ba8afce48fad4f4b2d0d357dc31e190482ce88
Added signature handling